test-clients/src/main/java/com/hedera/services/bdd/suites/suiterunner/TypedSuiteRunner.java

```diff
@@ -0,0 +1,108 @@
+package com.hedera.services.bdd.suites.suiterunner;
+
+/*-
+ * ‌
+ * Hedera Services Test Clients
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hedera.services.bdd.suites.SuiteRunner;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.hedera.services.bdd.suites.suiterunner.SuiteCategory.PERF_SUITES;
+import static com.hedera.services.bdd.suites.suiterunner.SuitesStore.getCategories;
+import static com.hedera.services.bdd.suites.suiterunner.SuitesStore.initialize;
+
+// TODO: implement dialogue before running the tests to display available options
```

Comment:
I'd suggest to not use a dialogue pattern in this file. All the necessary arguments should be passed down as command-line arguments.
Reason - we cannot provide input for the dialogue in the github actions vm.

Reply:
The dialogue pattern will be useful when we are running E2E tests locally. The developer will be aware of his options and the requirements for composing the argument string or array.

Reply:
I think this is overengineering tbh